src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +346,322 @@ void func_80B67540(EnSth* this, PlayState* play) {
+        if (STH_GET_SWAMP_BODY(&this->actor)) {
+            SkelAnime_InitFlex(play, &this->skelAnime, &gAhgSkel, &gEnSthBendDownAnim, this->jointTable,
+                               this->morphTable, AHG_LIMB_MAX);
+            Animation_PlayLoop(&this->skelAnime, &gEnSthBendDownAnim);
+            this->curAnimIndex = STH_ANIM_BENDING_DOWN;
             if (CHECK_WEEKEVENTREG(WEEKEVENTREG_34_10) || CHECK_WEEKEVENTREG(WEEKEVENTREG_34_20) ||
-                CHECK_WEEKEVENTREG(WEEKEVENTREG_34_40) || (Inventory_GetSkullTokenCount(play->sceneId) >= 30)) {
-                func_80B670A4(this, 3);
+                CHECK_WEEKEVENTREG(WEEKEVENTREG_34_40) ||
+                (Inventory_GetSkullTokenCount(play->sceneId) >= STH_SWAMP_SPIDER_TOKENS_REQUIRED)) {
+                EnSth_SwitchAnimation(this, STH_ANIM_WAIT);
             }
         } else {
-            SkelAnime_InitFlex(play, &this->skelAnime, &object_sth_Skel_0031F8, &ovl_En_Sth_Anim_003F50,
-                               this->jointTable, this->morphTable, 16);
-            Animation_PlayLoop(&this->skelAnime, &ovl_En_Sth_Anim_003F50);
+            SkelAnime_InitFlex(play, &this->skelAnime, &gSthSkel, &gEnSthSignalAnim, this->jointTable, this->morphTable,
+                               STH_LIMB_MAX);
+            Animation_PlayLoop(&this->skelAnime, &gEnSthSignalAnim);
         }
 
-        this->actor.update = func_80B680A8;
-        this->actor.draw = func_80B6849C;
+        this->actor.update = EnSth_Update;
+        this->actor.draw = EnSth_Draw;
 
-        switch (ENSTH_GET_F(&this->actor)) {
-            case ENSTH_F_3:
-                func_80B670A4(this, 4);
+        switch (STH_GET_TYPE(&this->actor)) {
+            case STH_TYPE_MOON_LOOKING:
+                EnSth_SwitchAnimation(this, STH_ANIM_LOOK_UP);
                 break;
 
-            case ENSTH_F_4:
+            case STH_TYPE_OCEANSIDE_SPIDER_HOUSE_GREET:
                 if (CHECK_WEEKEVENTREG(WEEKEVENTREG_13_80)) {
-                    func_80B670A4(this, 5);
+                    EnSth_SwitchAnimation(this, STH_ANIM_LOOK_AROUND);
                 } else {
-                    func_80B670A4(this, 5);
+                    EnSth_SwitchAnimation(this, STH_ANIM_LOOK_AROUND);
                 }
                 break;
 
-            case ENSTH_F_5:
-                func_80B670A4(this, 7);
+            case STH_TYPE_OCEANSIDE_SPIDER_HOUSE_PANIC:
+                EnSth_SwitchAnimation(this, STH_ANIM_PANIC);
                 break;
         }
 
-        if ((ENSTH_GET_F(&this->actor) == ENSTH_F_4) && (Inventory_GetSkullTokenCount(play->sceneId) < 30)) {
-            this->actor.update = func_80B67E20;
+        // not ready to appear yet
+        if ((STH_GET_TYPE(&this->actor) == STH_TYPE_OCEANSIDE_SPIDER_HOUSE_GREET) &&
+            (Inventory_GetSkullTokenCount(play->sceneId) < STH_OCEANSIDE_SPIDER_TOKENS_REQUIRED)) {
+            this->actor.update = EnSth_OceansideSpiderHouseWaitForTokens;
```

Comment:
Same here, although it's weird how many different main update functions there are